model/task/db.go

```diff
@@ -802,6 +802,105 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+type TaskBuildVariants struct {
+	Task          string   `bson:"_id"`
+	BuildVariants []string `bson:"build_variants"`
+}
+
+// FindUniqueBuildVariantNamesByTask returns TaskBuildVariants with a list of unique build variants
+func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) (*TaskBuildVariants, error) {
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			ProjectKey:     projectId,
+			DisplayNameKey: taskName,
+			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes}},
+		}}
+
+	group := bson.M{
+		"$group": bson.M{
+			"_id":                 taskName,
+			BuildVariantKey + "s": bson.M{"$addToSet": "$" + BuildVariantKey},
+		},
+	}
+	unwindAndSort := []bson.M{
+		{
+			"$unwind": "$build_variants",
+		},
+		{
+			"$sort": bson.M{
+				"build_variants": 1,
+			},
+		},
+		{
+			"$group": bson.M{
+				"_id":                 "$_id",
+				BuildVariantKey + "s": bson.M{"$push": "$" + BuildVariantKey + "s"},
+			},
+		},
+	}
+
+	pipeline = append(pipeline, group)
+	pipeline = append(pipeline, unwindAndSort...)
+
+	result := []TaskBuildVariants{}
```

Comment:
I'm not convinced we need this extra data structure; is there a way to just output the string list since that's all we need? If not, I think it's all we should return (and make the data structure private and internal to the function), since the caller doesn't need the task name given back to them.

Reply:
We need the data structure for the aggregation since the aggregation needs a return object. But i updated the function to return an array of strings.